src/hotspot/share/opto/output.cpp

```diff
@@ -1085,6 +1085,30 @@ void PhaseOutput::Process_OopMap_Node(MachNode *mach, int current_offset) {
           }
           scval = sv;
         }
+      } else if (obj_node->is_SafePointScalarMerge()) {
+        SafePointScalarMergeNode* smerge = obj_node->as_SafePointScalarMerge();
+        ObjectMergeValue* mv = (ObjectMergeValue*) sv_for_node_id(objs, smerge->_idx);
+
+        if (mv == NULL) {
```

Comment:
You should replace `NULL` with `nullptr` here and below. This also seems wrong here where you took the code from:
https://github.com/openjdk/jdk/blob/88378ed0584c7eb0849b6fc1e361fd8ea0698caf/src/hotspot/share/opto/output.cpp#L775-L796

On a separate note, the code looks almost identical. Could it be shared somehow?

Reply:
Thank you for reviewing @chhagedorn. I've converted the NULLs to nullptrs. However, I'll defer the refactoring of the identical code to a RFE - mainly because I'll have to backport the current patch and I'd like to keep it as minimal as possible. Please let me know if you disagree.

Reply:
That's perfectly fine, especially since the code for the `is_SafePointScalarObject()` case was already duplicated before. So, we could change both in one go in a follow-up RFE.